packages/@react-aria/button/src/useToggleButtonGroup.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  AnchorHTMLAttributes,
+  ButtonHTMLAttributes,
+  ElementType,
+  HTMLAttributes,
+  InputHTMLAttributes
+} from 'react';
+import {AriaLabelingProps, DOMAttributes, Orientation, RefObject} from '@react-types/shared';
+import {AriaToggleButtonGroupItemProps} from '@react-types/button';
+import {filterDOMProps, mergeProps} from '@react-aria/utils';
+import {ToggleButtonAria, useToggleButton} from './useToggleButton';
+import {ToggleGroupProps, ToggleGroupState, ToggleState} from '@react-stately/toggle';
+import {useToolbar} from '@react-aria/toolbar';
+
+export interface AriaToggleButtonGroupProps extends ToggleGroupProps, AriaLabelingProps {
+  /**
+   * The orientation of the the toggle button group.
+   * @default 'horizontal'
+   */
+  orientation?: Orientation
+}
+
+export interface ToggleButtonGroupAria {
+  /**
+   * Props for the toggle button group container.
+   */
+  groupProps: DOMAttributes
+}
+
+export function useToggleButtonGroup(props: AriaToggleButtonGroupProps, state: ToggleGroupState, ref: RefObject<HTMLElement | null>): ToggleButtonGroupAria {
+  let {isDisabled} = props;
+  let {toolbarProps} = useToolbar(props, ref);
+
+  return {
+    groupProps: mergeProps(toolbarProps, filterDOMProps(props, {labelable: true}), {
```

Comment:
seems like not. data attributes don't make it onto the element without this. Looks like in Toolbar we did the filterDOMProps in the RAC component instead of the hook. I forget which one we usually go with...